### PR TITLE
Add SoftGlobalScope flag

### LINKED ISSUE #11: Consistency with IJulia and NBInclude with the soft global scope?
First of all, I love the new soft global scope.  But is `NBInclude.jl` using it the same was as IJulia?

Here is my test case.  Create a cell with
```julia
n = 4
for i=1:2
    @show i
    n+= i
end
n
```

Then another, separate, code cell beneath it with just
```julia
n
```

If you run the two cells in Jupyter, it works great.  No globals required, and the second `n` correctly captures the modified value, as you need for any notebook style programming.

But if I run that same notebook with `NBInclude.jl`  it says
```julia
julia> @nbinclude("Untitled.ipynb")

ERROR: LoadError: UndefVarError: n not defined
Stacktrace:
 [1] top-level scope at /mnt/c/Users/jlperla/Documents/projects/Untitled.ipynb:In[4]:3 [inlined]
 [2] top-level scope at ./none:0
 [3] include_string(::Module, ::String, ::String) at ./loading.jl:1002
 [4] my_include_string(::Module, ::String, ::String, ::Nothing) at /home/jlperla/.julia/packages/NBInclude/YVOMm/src/NBInclude.jl:31
 [5] #nbinclude#1(::Bool, ::UnitRange{Int64}, ::Regex, ::typeof(identity), ::Function, ::Module, ::String) at /home/jlperla/.julia/packages/NBInclude/YVOMm/src/NBInclude.jl:83
 [6] nbinclude(::Module, ::String) at /home/jlperla/.julia/packages/NBInclude/YVOMm/src/NBInclude.jl:54
 [7] top-level scope at none:0
in expression starting at /mnt/c/Users/jlperla/Documents/projects/Untitled.ipynb:In[4]:2
```

-------

For my packages
```julia
(v1.0) pkg> st
    Status `~/.julia/environments/v1.0/Project.toml`
  [c52e3926] Atom v0.7.4
  [0c46a032] DifferentialEquations v5.3.0
  [31c24e10] Distributions v0.16.2
  [2fe49d83] Expectations v1.0.1
  [7073ff75] IJulia v1.10.0
  [a98d9a8b] Interpolations v0.8.0
  [0db19996] NBInclude v2.0.1
  [429524aa] Optim v0.17.0
  [d96e819e] Parameters v0.9.2
  [91a5bcdd] Plots v0.20.1
  [295af30f] Revise v0.7.3
```

(I tried to upload the notebook, but github won't let me)

### PR DESCRIPTION
**Pull request by arnavs**
What it says on the tin. I'm not 100% clear on the Travis/testing setup, so that may require some editing. Fixes #11.

cc: @jlperla

### CODE REVIEW

**Comment by jlperla** [comment]
Does `Pkg.test` work on your machine with this PR?

**Comment by arnavs** [comment]
Caught the error. My test without global scoping was looking for an `ErrorException`, of which `LoadError` is not a subset.

**Comment by arnavs** [comment]
(The Travis failures above are coming from the expected failures on 0.6 and 0.7, where this scoping rule is not broken).

**Comment by jlperla** [comment]
Should this feature and the tests be only run on >= v1.0 then?  No need to break the builds for v0.6 and v0.7 unnecessarily.  I think you can just check ` v"1.0" <= VERSION` in the test (and the code, if necessary).

**Comment by arnavs** [comment]
I'll give that a shot.

File: test/includes/scopetest.jl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+@testset "Scoping Tests" begin
+    @test_throws LoadError @nbinclude("scoping.ipynb")
+    @nbinclude("scoping.ipynb"; softscope = true)
+    @test a == 11
+end
```
Comment (stevengj) [inline test/includes/scopetest.jl]: I would put this test directly into runtests.jl

File: test/includes/scopetest.jl [hunk 2]
```diff
@@ -0,0 +1,5 @@
+@testset "Scoping Tests" begin
+    @test_throws LoadError @nbinclude("scoping.ipynb")
```
Comment (stevengj) [inline test/includes/scopetest.jl]: Put the `if VERSION >= v"1.0.0"` test on this line, but still do the `softscope=true` test on earlier versions.

File: test/Untitled.ipynb [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{
```
Comment (stevengj) [inline test/Untitled.ipynb]: Looks like this file got accidentally added?

File: src/NBInclude.jl [hunk 4]
```diff
@@ -48,7 +48,8 @@ function nbinclude(m::Module, path::AbstractString;
                    renumber::Bool=false,
                    counters = 1:typemax(Int),
                    regex::Regex = r"",
-                   anshook = identity)
+                   anshook = identity,
+                   softscope::Bool=false)
```
Comment (stevengj) [inline src/NBInclude.jl:50]: Documentation needs updating too.

File: test/includes/scopetest.jl (hunk 1 above)
Comment (arnavs) [reply test/includes/scopetest.jl]: Done.

File: test/includes/scopetest.jl (hunk 2 above)
Comment (arnavs) [reply test/includes/scopetest.jl]: Done.

File: test/Untitled.ipynb (hunk 3 above)
Comment (arnavs) [reply test/Untitled.ipynb]: Yup. Good catch.

File: src/NBInclude.jl (hunk 4 above)
Comment (arnavs) [reply src/NBInclude.jl:50]: Done.

**Comment by stevengj** [comment]
The docstrings in the code also needs an update.

**Comment by arnavs** [comment]
@stevengj I'm having trouble pulling your online changes into my fork. So I'll just duplicate them locally.

**Comment by arnavs** [comment]
OK @stevengj I think this is ready for your consumption. I've (a) updated the documentation and docstrings, (b) performed those changes on the tests, and (c) done the housekeeping on redundant files. The Travis build passes.

**Pull request merged**